src/components/implementation/no_interface/llbooter/boot_deps.h

```diff
@@ -105,6 +139,96 @@ boot_newcomp_sinv_alloc(spdid_t spdid)
 	}
 }
 
+/* TODO: possible to cos_defcompinfo_child_alloc if we somehow move allocations to one place */
+static void
+boot_newcomp_defcinfo_init(spdid_t spdid)
+{
+	struct cos_defcompinfo *defci     = cos_defcompinfo_curr_get();
+	struct cos_aep_info *   sched_aep = cos_sched_aep_get(defci);
+	struct cos_aep_info *   child_aep = cos_sched_aep_get(new_comp_cap_info[spdid].defci);
+	struct cos_compinfo *   child_ci  = cos_compinfo_get(new_comp_cap_info[spdid].defci);
+
+	child_aep->thd = cos_initthd_alloc(boot_info, child_ci->comp_cap);
+	assert(child_aep->thd);
+
+	if (new_comp_cap_info[spdid].is_sched) {
+		child_aep->tc = cos_tcap_alloc(boot_info);
+		assert(child_aep->tc);
+
+		child_aep->rcv = cos_arcv_alloc(boot_info, child_aep->thd, child_aep->tc, boot_info->comp_cap, sched_aep->rcv);
+		assert(child_aep->rcv);
+	} else {
+		child_aep->tc  = sched_aep->tc;
+		child_aep->rcv = sched_aep->rcv;
+	}
+
+	child_aep->fn   = NULL;
+	child_aep->data = NULL;
+}
+
+static void
+boot_newcomp_init_caps(spdid_t spdid)
+{
+	struct cos_compinfo  *ci    = new_comp_cap_info[spdid].compinfo;
+	struct comp_cap_info *capci = &new_comp_cap_info[spdid];
+	int ret, i;
+	
+	/* FIXME: not everyone should have it. but for now, because getting cpu cycles uses this */
```

Comment:
Don't only schedulers need this?

Reply:
Even non-scheduling components may need to use this. I can think of: 
1. Application needing to compute some deadlines or doing benchmarks. `rdtsc` and convert to usecs.
2. Applications calling down to their schedulers to set `timeout` in `xxxx_thd_block_timeout`. 

I can't fully understand why this is just a scheduler need.